sdk/src/main/kotlin/io/rover/sdk/ui/blocks/poll/ViewTextPoll.kt

```diff
@@ -1,47 +1,116 @@
 package io.rover.sdk.ui.blocks.poll
 
+import android.graphics.Typeface
+import android.support.v4.view.ViewCompat
+import android.support.v7.widget.AppCompatTextView
+import android.view.ViewGroup
 import android.widget.LinearLayout
+import io.rover.sdk.data.domain.QuestionStyle
 import io.rover.sdk.data.domain.TextPollBlock
-import io.rover.sdk.platform.button
-import io.rover.sdk.platform.setDimens
+import io.rover.sdk.platform.addView
+import io.rover.sdk.platform.mapToFont
+import io.rover.sdk.platform.optionView
+import io.rover.sdk.platform.setupLayoutParams
 import io.rover.sdk.platform.textView
+import io.rover.sdk.streams.androidLifecycleDispose
+import io.rover.sdk.streams.subscribe
+import io.rover.sdk.ui.asAndroidColor
+import io.rover.sdk.ui.blocks.concerns.background.BackgroundViewModelInterface
+import io.rover.sdk.ui.blocks.concerns.background.createBackgroundDrawable
 import io.rover.sdk.ui.concerns.MeasuredBindableView
+import io.rover.sdk.ui.concerns.MeasuredSize
 import io.rover.sdk.ui.concerns.ViewModelBinding
-import io.rover.sdk.ui.dpAsPx
 
 internal class ViewTextPoll(override val view: LinearLayout) : ViewTextPollInterface {
-    override var viewModelBinding: MeasuredBindableView.Binding<TextPollViewModelInterface>? by ViewModelBinding { binding, _ ->
-        binding?.viewModel?.textPollBlock?.let {
-            createViews(it)
+
+    companion object {
+        private const val MAX_OPTIONS_AMOUNT = 4
+    }
+
+    private val optionIds = (0 until MAX_OPTIONS_AMOUNT).map { ViewCompat.generateViewId() }
```

Comment:
I have a dumb question: why not just hold on to references to the View objects, rather than wanting to de-reference  them by ID?

Reply:
I've changed it the hold references to the View objects, I didn't want to have to litter the code with nullability related logic at first but i've avoided this now through `lateinit` for views that can't be instantiated upon instantiation of the `ViewTextPoll`.